pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -82,9 +123,11 @@ func (e *HookExecutor) Execute(hook *deployapi.LifecycleHook, deployment *kapi.R
 	// Retry failures are treated the same as Abort.
 	switch hook.FailurePolicy {
 	case deployapi.LifecycleHookFailurePolicyAbort, deployapi.LifecycleHookFailurePolicyRetry:
-		return fmt.Errorf("%s hook failed, aborting deployment: %s", label, err)
+		e.event(deployment, kapi.EventTypeWarning, "Failed", fmt.Sprintf("The %s hook failed: %v, aborting deployment %s/%s", label, err, deployment.Namespace, deployment.Name))
```

Comment:
My main concern with having these events as part of the replication controller is that then we will end up with fragmented information, because we already moved all of the other events regarding the deployment process on the config: https://github.com/openshift/origin/pull/7830

It is also consistent with what's happening upstream:
https://github.com/kubernetes/kubernetes/blob/b6f966f8f5b4903342fa0aec6f63b1fec3c030ea/pkg/controller/deployment/deployment_controller.go#L1368

@mfojtik by pod creation you mean the hook? We should move this too on the config if that's not the case currently.